misk-hibernate/src/test/kotlin/misk/hibernate/ReflectionQueryFactoryTest.kt

```diff
@@ -236,6 +240,119 @@ class ReflectionQueryFactoryTest {
     }
   }
 
+  @Test
+  fun runtimeQueryConstraints() {
+    val m1 = NameAndReleaseDate("Rocky 1", LocalDate.of(2018, 1, 1))
+    val m2 = NameAndReleaseDate("Rocky 2", LocalDate.of(2018, 1, 2))
+    val m3 = NameAndReleaseDate("Rocky 3", LocalDate.of(2018, 1, 3))
+    val m99 = NameAndReleaseDate("Rocky 99", null)
+
+    transacter.allowCowrites().transaction { session ->
+      session.save(DbMovie(m1.name, m1.releaseDate))
+      session.save(DbMovie(m2.name, m2.releaseDate))
+      session.save(DbMovie(m3.name, m3.releaseDate))
+      session.save(DbMovie(m99.name, m99.releaseDate))
+
+      val queryObjectResult = queryFactory.newQuery<OperatorsMovieQuery>()
+          .releaseDateLessThan(m3.releaseDate)
+          .allowFullScatter().allowTableScan()
+          .listAsNameAndReleaseDate(session)
+
+      // Should contain the same items as running the query with a runtime constraint.
+      assertThat(queryFactory.newQuery<OperatorsMovieQuery>()
+          .apply { addConstraint("release_date", LT, m3.releaseDate) }
+          .allowFullScatter().allowTableScan()
+          .listAsNameAndReleaseDate(session))
+          .containsExactlyInAnyOrderElementsOf(queryObjectResult)
```

Comment:
should order be consistent?

Reply:
Here I don't believe we care. We aren't setting a sort. If there is a sort we do.